client/src/pages/ProductDetails.tsx

```diff
@@ -27,6 +27,24 @@ export function ProductDetails() {
     loadProduct(Number(productId));
   }, [productId]);
 
+  async function handleAddToCart(
+    event: FormEvent<HTMLFormElement>
+  ): Promise<void> {
+    try {
+      event.preventDefault();
+      if (!product) throw new Error('you broke something');
+
+      const formData = new FormData(event.currentTarget);
+      const formValues = Object.fromEntries(formData);
+      console.log('formValues:', formValues);
+      addToCart(product, +formValues.sizes);
```

Comment:
I think you need to `await` this. That's why you get the "added to cart" alert even when there's an error.